### PR TITLE
Replace service.yaml with the correct contents

### LINKED ISSUE #1312: Failed to set up the webhook in Github for gitwebhook-go sample
I followed the tutorial at https://github.com/knative/docs/tree/master/docs/serving/samples/gitwebhook-go.

I used the GKE as the Kube backend. After configuring the domain and the static IP, I still failed to create a valid webhook available to receive the payload successfully. 

SSL validation option is only available when we use HTTPS protocol, like https://\<domain\>, and it is not available for HTTP protocol, like http://\<domain\> as the payload URL.

No matter I used https://\<domain\> or http://\<domain\>, I failed to get successful response from the webhook, when I delivered any payload to the URL http://\<domain\>.

It popped up with "405 method not allowed", when I run "curl -H "Host: gitwebhook.default.\<domain\>.com" http://${IP_ADDRESS}"

### PR DESCRIPTION
**Pull request by houshengbo**
Fixes #1312

## Proposed Changes

- Replace service.yaml with the correct contents
- Add the requirement for a public domain name for the service

### CODE REVIEW

File: docs/serving/samples/gitwebhook-go/README.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ webhook.
   and a Docker Hub account configured (we'll use it for a container registry).
 - An account on [GitHub](https://github.com) with read/write access to a
   repository.
+- A public domain name available over the internet for your service to run this
+  example, if you use the repository in github.com to send a payload to your
+  webhook service. Check the guidance on how to get one for your service
```
Comment (gyliu513) [inline docs/serving/samples/gitwebhook-go/README.md]: Can you help add a link to the `guidance`?

Comment (houshengbo) [reply docs/serving/samples/gitwebhook-go/README.md]: Each cloud provider should have its own guidance. I have not investigated all of them.

File: docs/serving/samples/gitwebhook-go/README.md [hunk 2]
```diff
@@ -10,6 +10,10 @@ webhook.
   and a Docker Hub account configured (we'll use it for a container registry).
 - An account on [GitHub](https://github.com) with read/write access to a
   repository.
+- A public domain name available over the internet for your service to run this
+  example, if you use the repository in github.com to send a payload to your
```
Comment (gyliu513) [inline docs/serving/samples/gitwebhook-go/README.md]: s/github.com/[GitHub](https://github.com)

File: docs/serving/samples/gitwebhook-go/README.md (hunk 1 above)
Comment (gyliu513) [reply docs/serving/samples/gitwebhook-go/README.md]: make sense, thanks.

File: docs/serving/samples/gitwebhook-go/README.md (hunk 2 above)
Comment (houshengbo) [reply docs/serving/samples/gitwebhook-go/README.md]: done

File: docs/serving/samples/gitwebhook-go/README.md (hunk 1 above)
Comment (sebgoa) [reply docs/serving/samples/gitwebhook-go/README.md]: Step 6 further down actually points to the doc about configuring a custom domain and assigning a static IP to it. I think we should bring that information to the sentence that you just added.

**Comment by RichieEscarez** [comment]
I did a quick edit to make clear that owning a domain is a prereq and also fixed some steps, links, and PR #1202 

PTAL to ensure technical accuracy and then I can approve/merge

File: docs/serving/samples/gitwebhook-go/README.md [hunk 3]
```diff
@@ -3,35 +3,45 @@ webhook.
...
+- A Kubernetes cluster running with the following:
+  - Knative Serving must be installed. For details about
+    setting up a Knative cluster, see the
+    [installation guides](../../../install/README.md).
+  - Your Knative cluster must be 
+    [configured to use your custom domain](../../using-a-custom-domain.md).
+  - You must ensure that your Knative cluster uses a static IP address:
+    - For Google Kubernetes Engine, see 
+      [assigning a static IP address](../../gke-assigning-static-ip-address.md).
+    - For other cloud provider, refer to your provider's documentation.
```
Comment (RichieEscarez) [inline docs/serving/samples/gitwebhook-go/README.md]:
```suggestion
    - For other cloud providers, refer to your provider's documentation.
```

Comment (houshengbo) [reply docs/serving/samples/gitwebhook-go/README.md]: done

File: docs/serving/samples/gitwebhook-go/README.md [hunk 4]
```diff
@@ -3,35 +3,45 @@ webhook.
...
-   ```shell
-   git clone -b "release-0.6" https://github.com/knative/docs knative-docs
-   cd knative-docs/serving/samples/gitwebhook-go
-   ```
+    ```shell
+    git clone -b "release-0.6" https://github.com/knative/docs knative-docs
+    cd knative-docs/serving/samples/gitwebhook-go
+    ```
 
 1. Use Docker to build a container image for this service. Replace `username`
```
Comment (RichieEscarez) [inline docs/serving/samples/gitwebhook-go/README.md]:
```suggestion
1. Use Docker to build a container image for this service. Replace `{DOCKER_HUB_USERNAME}`
```

File: docs/serving/samples/gitwebhook-go/README.md [hunk 5]
```diff
@@ -3,35 +3,45 @@ webhook.
...
+    docker build -t ${DOCKER_HUB_USERNAME}/gitwebhook-go .
 
-   # Push the container to the registry
-   docker push ${DOCKER_HUB_USERNAME}/gitwebhook-go
-   ```
+    # Push the container to the registry
+    docker push ${DOCKER_HUB_USERNAME}/gitwebhook-go
+    ```
 
 1. Create a secret that holds two values from GitHub, a personal access token
```
Comment (RichieEscarez) [inline docs/serving/samples/gitwebhook-go/README.md]:
```suggestion
1. Create a secret that holds two values from GitHub:
```

File: docs/serving/samples/gitwebhook-go/README.md [hunk 6]
```diff
@@ -3,35 +3,45 @@ webhook.
...
    used to make API requests to GitHub, and a webhook secret, used to validate
```
Comment (RichieEscarez) [inline docs/serving/samples/gitwebhook-go/README.md]:
```suggestion
    -  A personal access token that you will use to make API requests to GitHub
       - Ensure that you grant `read/write` permission in the repo for that personal access token.
```

File: docs/serving/samples/gitwebhook-go/README.md [hunk 7]
```diff
@@ -41,84 +51,110 @@ webhook.
       [create a personal access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/).
```
Comment (RichieEscarez) [inline docs/serving/samples/gitwebhook-go/README.md]:
```suggestion
    - A webhook secret that you will used to validate requests.
```

File: docs/serving/samples/gitwebhook-go/README.md [hunk 8]
```diff
@@ -41,84 +51,110 @@ webhook.
...
       Ensure to grant the `repo` permission to give `read/write` access to the
```
Comment (RichieEscarez) [inline docs/serving/samples/gitwebhook-go/README.md]:
```suggestion
```

File: docs/serving/samples/gitwebhook-go/README.md [hunk 9]
```diff
@@ -41,84 +51,110 @@ webhook.
...
       personal access token.
```
Comment (RichieEscarez) [inline docs/serving/samples/gitwebhook-go/README.md]:
```suggestion
```

**Comment by RichieEscarez** [comment]
A few more edits but otherwise this looks good to me. Ill mark 'approved' once my suggestions are committed.

Can @sebgoa or @gyliu513 provide the LGTM?

**Comment by gyliu513** [comment]
/lgtm

Thanks @houshengbo

**Pull request merged**